tests/integration_python/test_run_cli.py

```diff
@@ -1356,3 +1356,98 @@ def test_task_caching_with_multiple_inputs_args(pixi: Path, tmp_pixi_workspace:
             "cache hit",
         ],
     )
+
+
+# Run with environment variable and sort the priority
+# variable task.env > activation.env > activation.scripts > activation scripts of dependencies > outside environment variable
+def test_run_with_environment_variable_priority(
+    pixi: Path, tmp_pixi_workspace: Path, dummy_channel_1: str
+) -> None:
+    manifest = tmp_pixi_workspace.joinpath("pixi.toml")
+    script_manifest = tmp_pixi_workspace.joinpath("env_setup.sh")
+    toml = f"""
+    [project]
+    name = "test"
+    channels = ["{dummy_channel_1}"]
+    platforms = ["linux-64", "osx-64", "osx-arm64", "win-64"]
+    [activation.env]
+    MY_ENV = "test123"
+    [activation]
+    scripts = ["env_setup.sh"]
+    [tasks.task]
+    cmd = "echo $MY_ENV"
+    [tasks.foo]
+    cmd = "echo $MY_ENV"
+    [tasks.foobar]
+    cmd = "echo $FOO_PATH"
+    [tasks.task.env]
+    MY_ENV = "test456"
+    [dependencies]
+    pixi-foobar = "*"
+    """
+    test_script_file = """
+    #!/bin/bash
```

Comment:
This will only work on Unix.

Something like this should work:

```toml
[target.unix.activation]
scripts = ["scripts/activate.sh"]
[target.win-64.activation]
scripts = ["scripts/activate.bat"]
```

Reply:
Good catch!

Reply:
> This will only work on Unix.
> 
> Something like this should work:
> 
> ```toml
> [target.unix.activation]
> scripts = ["scripts/activate.sh"]
> [target.win-64.activation]
> scripts = ["scripts/activate.bat"]
> ```

good catch!